pkg/eosclient/eosgrpc/eosgrpc.go

```diff
@@ -875,71 +936,100 @@ func (c *Client) List(ctx context.Context, uid, gid, dpath string) ([]*eosclient
 	for {
 		rsp, err := resp.Recv()
 		if err != nil {
-			if err == io.EOF {
+			log.Debug().Err(err).Str("func", "List").Str("path", dpath).Str("got err from EOS", err.Error()).Msg("grpc response")
+			if err == io.EOF || i > 0 {
+				log.Debug().Str("path", dpath).Int("nitems", i-1).Msg("OK, no more items")
 				return mylst, nil
 			}
 
-			log.Warn().Err(err).Str("path", dpath).Str("err", err.Error())
-
 			return nil, err
 		}
 
 		if rsp == nil {
-			log.Warn().Err(err).Str("path", dpath).Str("err", "rsp is nil")
+			log.Warn().Err(err).Str("func", "List").Str("path", dpath).Str("err", "rsp is nil").Msg("grpc response")
 			return nil, errtypes.NotFound(dpath)
 		}
 
-		log.Debug().Str("path", dpath).Str("item resp:", fmt.Sprintf("%#v", rsp)).Msg("grpc response")
+		i++
+		if i == 1 {
+			log.Debug().Str("func", "List").Str("path", dpath).Str("skipping item resp:", fmt.Sprintf("%#v", rsp)).Msg("grpc response")
+			continue
+		}
+		log.Debug().Str("func", "List").Str("path", dpath).Str("item resp:", fmt.Sprintf("%#v", rsp)).Msg("grpc response")
 
 		myitem, err := c.grpcMDResponseToFileInfo(rsp, dpath)
 		if err != nil {
-			log.Warn().Err(err).Str("path", dpath).Str("could not convert item:", fmt.Sprintf("%#v", rsp)).Str("err:", err.Error()).Msg("")
+			log.Warn().Err(err).Str("func", "List").Str("path", dpath).Str("could not convert item:", fmt.Sprintf("%#v", rsp)).Str("err", err.Error()).Msg("")
 
 			return nil, err
 		}
 
-		i++
-		if i == 1 {
-			continue
-		}
 		mylst = append(mylst, myitem)
 	}
+
 }
 
-// Read reads a file from the mgm
+// Read reads a file from the mgm. It may or may not create a temp copy in the local filesystem
 func (c *Client) Read(ctx context.Context, uid, gid, path string) (io.ReadCloser, error) {
-	rand := "eosread-" + uuid.New().String()
-	localTarget := fmt.Sprintf("%s/%s", c.opt.CacheDirectory, rand)
-	defer os.RemoveAll(localTarget)
+	log := appctx.GetLogger(ctx)
+	log.Info().Str("func", "Read").Str("uid,gid", uid+","+gid).Str("path", path).Msg("")
 
-	xrdPath := fmt.Sprintf("%s//%s", c.opt.URL, path)
-	cmd := exec.CommandContext(ctx, c.opt.XrdcopyBinary, "--nopbar", "--silent", "-f", xrdPath, localTarget, fmt.Sprintf("-OSeos.ruid=%s&eos.rgid=%s", uid, gid))
-	if _, _, err := c.execute(ctx, cmd); err != nil {
-		return nil, err
-	}
-	return os.Open(localTarget)
-}
+	var localTarget string
+	//rand := "eosread-" + uuid.New().String()
+	//localTarget := fmt.Sprintf("%s/%s", c.opt.CacheDirectory, rand)
+	//defer os.RemoveAll(localTarget)
 
-// Write writes a file to the mgm
-func (c *Client) Write(ctx context.Context, uid, gid, path string, stream io.ReadCloser) error {
-	fd, err := ioutil.TempFile(c.opt.CacheDirectory, "eoswrite-")
-	if err != nil {
-		return err
-	}
-	defer fd.Close()
-	defer os.RemoveAll(fd.Name())
+	var localfile io.WriteCloser
+	localfile = nil
```

Comment:
This can be removed. It's nil by default

Reply:
This is temporary code, until we understand what to do there